R/summary_funs.R

```diff
@@ -259,141 +233,178 @@ weighted_summary_means <- function(y_wobs_test, family, wdraws, mu, dis, cl_ref,
-                          names = FALSE, na.rm = TRUE)
+      # simple transformation of mse
+      value <- sqrt(mse_e) - ifelse(is.null(summaries_baseline), 0, sqrt(mse_b))
+      # the first-order Taylor approximation of the variance
+      value_se <- sqrt(value_se^2 / mse_e / 4)
+    } else if (stat == "R2") {
+      # simple transformation of mse
+      mse_y <- mean(wobs * (mean(y) - y)^2)
+      value <- 1 - mse_e / mse_y - ifelse(is.null(summaries_baseline), 0, 1 - mse_b / mse_y)
+      # the first-order Taylor approximation of the variance
+      var_mse_y <- .weighted_sd((mean(y) - y)^2, wobs)^2 / n_full
+      if (n_loo < n_full) {
+        mse_e_fast <- mean(wobs * (summaries_fast$mu - y)^2)
+        if (is.null(summaries_baseline)) {
+          srs_diffe <-
+            .srs_diff_est_w(y_approx = ((summaries_fast$mu - y)^2 - mse_e_fast) *
+                              ((mean(y) - y)^2 - mse_y),
+                            y = (((mu - y)^2 - mse_e) *
+                                   ((mean(y) - y)^2 - mse_y))[loo_inds],
+                            y_idx = loo_inds,
+                            w = wobs)
+        } else {
+          srs_diffe <-
+            .srs_diff_est_w(y_approx = ((summaries_fast$mu - y)^2 - mse_e_fast -
+                                          ((mu_baseline - y)^2 - mse_b)) *
+                              ((mean(y) - y)^2 - mse_y),
+                            y = (((mu - y)^2 - mse_e -
+                                    ((mu_baseline - y)^2 - mse_b)) *
+                                   ((mean(y) - y)^2 - mse_y))[loo_inds],
+                            y_idx = loo_inds,
+                            w = wobs)
+        }
+        cov_mse_e_y <- srs_diffe$y_hat / n_full^2
       } else {
-        value <- sqrt(mean(wcv * (mu - y)^2, na.rm = TRUE))
-        value.bootstrap <- bootstrap(
-          (mu - y)^2,
-          function(resid2) {
-            sqrt(mean(wcv * resid2, na.rm = TRUE))
-          },
-          ...
-        )
-        value.se <- sd(value.bootstrap)
-        lq_uq <- quantile(value.bootstrap,
-                          probs = c(alpha_half, one_minus_alpha_half),
-                          names = FALSE, na.rm = TRUE)
+        if (is.null(summaries_baseline)) {
+          cov_mse_e_y <- mean(wobs * ((mu - y)^2 - mse_e) *
+                                ((mean(y) - y)^2 - mse_y)) / n_full
+        } else {
+          cov_mse_e_y <- mean(wobs * ((mu - y)^2 - mse_e -
+                                        ((mu_baseline - y)^2 - mse_b)) *
+                                ((mean(y) - y)^2 - mse_y)) / n_full
+        }
       }
+      # part of delta se comes automatically via mse
+      if (!is.null(summaries_baseline)) {
+        # delta=TRUE
+        mse_e <- mse_e - mse_b
+      }
+      value_se <- sqrt((value_se^2 -
+                          2 * mse_e / mse_y * cov_mse_e_y +
+                          (mse_e / mse_y)^2 * var_mse_y) / mse_y^2)
```

Comment:
If my understanding from https://github.com/stan-dev/loo/pull/205#issuecomment-2325238936 is correct, then I think we would need a trivariate delta method in the `!is.null(summaries_baseline)` case (because `mse_b` comes in, too). I haven't checked whether such a trivariate delta method would give the same formula as used here. Have you checked this?